addons/mail/models/mail_thread.py

```diff
@@ -89,12 +89,6 @@ class MailThread(models.AbstractModel):
         'mail.message', 'res_id', string='Messages',
         domain=lambda self: [('message_type', '!=', 'user_notification')], auto_join=True)
     has_message = fields.Boolean(compute="_compute_has_message", search="_search_has_message", store=False)
-    message_unread = fields.Boolean(
```

Comment:
Why ? I use it quite frequently. Not sure what you try to gain but I think this kind of fields cost nothing and can be used by people on a daily basis.

Reply:
Maybe you'll have to help me understand how it can be used then, as far as I understand the compute does a right-join on channel_partner on channel_id which is a mail.channel, so from my understanding these fields can only ever be False and 0 respectively for non-channel mail.threads.

Although now that I look at it, since we never check that the model is mail.channel, we can probably accidentally select a channel_partner that's completely unrelated to the thread we're interested in, and read an incorrect value for both of these fields because there happens to be a channel_partner for the mail channel with the same id as the thread.

Reply:
Seems actually what is called "Message Unread" in interface is using "message_needaction" . My bad, indeed current message_unread makes no sense for standard documents.

I would suggest to actually rename ``message_needaction`` to ``message_unread`` so that naming makes sense. While you are cleaning that part, let us maybe benefits from a renaming :) .

Reply:
We're sort of getting into a naming spaghetti bowl at this point. In the UI, we have a button called "mark as read" that calls a method named "set_message_done" which removes the "needaction" from the message. On the other hand, channel messages can be "seen" (last_seen_message_id etc) but we want to call the UI button "Mark as unread" because that's what it's called in every other messaging application, so we have this asymmetry between mark read and mark unread which don't represent the same "read" concept at all.

I don't mind cleaning this up to make the naming consistent across the board once the rest of this PR has been validated from a technical standpoint.

Reply:
@sdegueldre Well python models are the core of Odoo. I think it is more important to begin with a clear naming for fields, then gradually propagate it. Needaction is an old name coming from FP that meant about nothing. This is linked to unread notifications, so message_unread makes more sense (and 'done' or 'needaction' are wrong naming indeed). Renaming methods and frontend can be done gradually in my opinion. As you prefer anyway, it is your PR :) . Just pointing it out while reading the diff.